porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller_test.go

```diff
@@ -1134,3 +1135,409 @@ spec:
 		})
 	}
 }
+
+func TestEnsureKRMFunctions(t *testing.T) {
+	pvBase := `
+apiVersion: config.porch.kpt.dev
+kind: PackageVariant
+metadata:
+  name: my-pv
+  uid: pv-uid
+spec:
+  upstream:
+    repo: blueprints
+    package: foo
+    revision: v1
+  downstream:
+    repo: deployments
+    package: bar
+  pipeline:
+`[1:]
```

Comment:
Yes normally strings.TrimSpace works for this. Unfortunately it does not work in our case, as the second line is indented with spaces. This will result in trimming out the spaces of that line:


```Go
func Test(t *testing.T) {
	s := `
  second-line indented:
    more indented:
`
	fmt.Println(strings.TrimSpace(s))
	fmt.Println("------")
	fmt.Println(s[1:])
}

// will produce
second-line indented:
    more indented:
------
  second-line indented:
    more indented:
```

But I am also going to look into the idea of your other comment to make use of kyaml, which maybe makes this obsolete all together.

Reply:
Not a blocker for this PR. It is just a bit difficult to read with the indentation.